fhirr4/compiler-plugin/src/main/java/io/ballerinax/fhirr4/compiler/FHIRResourceValidator.java

```diff
@@ -60,10 +60,11 @@ public static void extractInputParamTypeAndValidate(SyntaxNodeAnalysisContext ct
             return;
         }
         TypeSymbol firstParamType = parametersOptional.get().get(0).typeDescriptor();
-        if (!(firstParamType.getName().get().equals(Constants.FHIR_CONTEXT)
-                && firstParamType.getModule().get().getName().get().equals(Constants.HEALTHCARE_PKG))) {
-            updateDiagnostic(ctx, paramLocation, FHIRDiagnosticCodes.FHIR_103,
-                             firstParamType.getName().get());
+        String firstParamSignature = firstParamType.signature();
+        String firstParamName = firstParamType.getName().orElse("");
+        String firstParamModuleName = firstParamType.getModule().flatMap(Symbol::getName).orElse("");
+        if (!(firstParamName.equals(Constants.FHIR_CONTEXT) && firstParamModuleName.equals(Constants.HEALTHCARE_PKG))) {
```

Comment:
without optional we should be able achieve this using, will remove additional if checks

f (!(Constants.FHIR_CONTEXT.equals(firstParamName) && Constants.HEALTHCARE_PKG.equals(firstParamModuleName))) {